### PR TITLE
add lidar driver of surestar from neolix

### PR DESCRIPTION
**Pull request by zyc0222**

### CODE REVIEW

**Review by storypku — CHANGES_REQUESTED** [review]
Please remove `.DS_Store` files and make sure all c/c++ headers were formated with `scripts/clang-format.sh`. And all review suggestions.

File: modules/drivers/lidar_surestar/include/component/component_convert.h [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+namespace autobot {
+namespace drivers {
+namespace surestar {
+using apollo::cyber::Component;
+class CompSurestarConvert
+    : public Component<adu::common::sensor::Surestar::SurestarScan> {
+ public:
+  bool Init() {
+    // read config file
+    cybertron::proto::SurestarConfig config;
```
Comment (storypku) [inline modules/drivers/lidar_surestar/include/component/component_convert.h:42]: Maybe `cybertron` should be renamed to `cyber` ?

File: modules/drivers/lidar_surestar/include/driver/driver.h [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+namespace drivers {
+namespace surestar {
+
+class SurestarDriver {
+ public:
+  SurestarDriver();
+  virtual ~SurestarDriver() {}
+
+  virtual bool poll(
+      const std::shared_ptr<adu::common::sensor::Surestar::SurestarScan>&
```
Comment (storypku) [inline modules/drivers/lidar_surestar/include/driver/driver.h:39]: And `adu`.

File: modules/drivers/lidar_surestar/src/lib/socket_input.cpp [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+#include <arpa/inet.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <poll.h>
+#include <sys/file.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+namespace autobot {
```
Comment (storypku) [inline modules/drivers/lidar_surestar/src/lib/socket_input.cpp:27]: Please follow Apollo's convention on namespace naming. Thanks!

File: modules/drivers/lidar_surestar/BUILD [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    deps = [":driver"],
+)
+
+cc_library(
+    name = "driver",
+    srcs = glob(
+        ["src/**/*.cpp"]
+    ),
+    hdrs = glob(
+        ["include/**/*.h"]
```
Comment (storypku) [inline modules/drivers/lidar_surestar/BUILD:18]: Try to fine grain Bazel build target. Follow Bazel Best Practices [here](https://docs.bazel.build/versions/master/bazel-and-cpp.html#best-practices).

**Review by storypku — CHANGES_REQUESTED** [review]

**Comment by Capri2014** [comment]
and why again check into 5.5 first? ci is not enabled for this branch.

**Comment by storypku** [comment]
> and why again check into 5.5 first? ci is not enabled for this branch.

Agree.

File: modules/drivers/lidar_surestar/include/component/component_convert.h (hunk 1 above)
Comment (zyc0222) [reply modules/drivers/lidar_surestar/include/component/component_convert.h]: All namespace has been renamed

File: modules/drivers/lidar_surestar/BUILD (hunk 4 above)
Comment (zyc0222) [reply modules/drivers/lidar_surestar/BUILD]: BUILD is refined accordingly

**Pull request closed without merging**